diff --git a/cpp/src/replace/clamp.cu b/cpp/src/replace/clamp.cu
index bf2ae63a99..b4f28f65a6 100644
--- a/cpp/src/replace/clamp.cu
+++ b/cpp/src/replace/clamp.cu
@@ -108,10 +108,8 @@ std::unique_ptr<cudf::column> clamp_string_column(strings_column_view const& inp
     return bytes;
   };
 
-  auto offset_and_char =
+  auto [offsets_column, chars_column] =
     form_offsets_and_char_column(d_input, null_count, offsets_transformer, stream, mr);
-  auto offsets_column(std::move(offset_and_char.first));
-  auto chars_column(std::move(offset_and_char.second));
 
   auto d_offsets = offsets_column->view().template data<size_type>();
   auto d_chars   = chars_column->mutable_view().template data<char>();
